src/backend/export/outputVendors/ynab/ynab.ts

```diff
@@ -75,16 +75,25 @@ function getTransactions(startDate: Date): Promise<ynab.TransactionsResponse> {
   return ynabAPI!.transactions.getTransactions(ynabConfig!.options.budgetId, moment(startDate).format(YNAB_DATE_FORMAT));
 }
 
+export function getPayeeName(transaction: EnrichedTransaction, payeeNameMaxLength: number = 50) {
+  // Specific case for Bank Hapoalim - Where we can extract the payee name from the memo. Note the "." is added
+  // from the israeli-bank-scrapers project, and not from the bank itself
+  if (transaction.memo?.startsWith('לטובת') && transaction.memo.includes('עבור')) {
+    return transaction.memo.slice(7, transaction.memo.indexOf('עבור') - 2);
+  }
+  return transaction.description.substring(0, payeeNameMaxLength);
+}
+
 function convertTransactionToYnabFormat(originalTransaction: EnrichedTransaction): ynab.SaveTransaction {
-  const payeeNameMaxLength = ynabConfig!.options.maxPayeeNameLength || 50;
   const amount = Math.round(originalTransaction.chargedAmount * 1000);
   const date = convertTimestampToYnabDateFormat(originalTransaction);
+
   return {
     account_id: getYnabAccountIdByAccountNumberFromTransaction(originalTransaction.accountNumber),
     date, // "2019-01-17",
     amount,
     // "payee_id": "string",
-    payee_name: originalTransaction.description.substring(0, payeeNameMaxLength),
+    payee_name: getPayeeName(originalTransaction, ynabConfig!.options.maxPayeeNameLength || 50),
```

Comment:
Isn't the || 50 redundant now that you made 50 the default value of the parameter?

Reply:
I'm not a js dev so bear with me,

if `ynabConfig!.options.maxPayeeNameLength` is undefined, won't `getPayeeName` get the value `undefined` or `null` or whatever value `maxPayeeNameLength` has?

Reply:
In general yes, but you are calling the getPayeeName function and passing this as the second parameter.
For this parameter you defined a default value of 50
`export function getPayeeName(transaction: EnrichedTransaction, payeeNameMaxLength: number = 50) {`

So if you don't pass a parameter or pass undefined (which should be the value of this if no one assigned any value to id) it will get the default value

See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters